File: ClarkSnacks.MTS.Web/src/app/models/inspection.ts
import { Lot } from './lot';

export class Inspection {
  lots: InspectionLot[];
  supplierId: number;
  dateReceived: string;
  bolShipmentNumber: string;
  isApprovedSupplier: boolean;
  questions: InspectionQuestions;
  resultId: number;
  dispositionId: number;
  deviationNumber: string;
  holdStatusId: number;
  measuresToPreventRelease: string;
  inspectedById: number
}

export class InspectionLot {
    id: number;
    itemId: number;
    lotNumber: string;
    quantity: number;
    comment: string;
    createdByUserId: number;
}

export class InspectionItem {
    id: number;
    itemId: number;
    inspectionLots: InspectionLot[] = [];
    totalQuantity: number;
}

export class InspectionQuestions {

  // Cartons
  cartonQuestion1: number;
  cartonQuestion1Comment: string;
  cartonQuestion2: number;
  cartonQuestion2Comment: string;
  cartonQuestion3: number;
  cartonQuestion3Comment: string;
  cartonQuestion4: number;
  cartonQuestion4Comment: string;
  cartonQuestion5: number;
  cartonQuestion5Comment: string;;
  cartonQuestion6: number;
  cartonQuestion6Comment: string;
  cartonQuestion7: number;
  cartonQuestion7Comment: string;
  cartonQuestion8: number;
  cartonQuestion8Comment: string;

  // Bags
  bagQuestion1: number;
  bagQuestion1Comment: string;
  bagQuestion2: number;
  bagQuestion2Comment: string;
  bagQuestion3: number;
  bagQuestion3Comment: string;
  bagQuestion4: number;
  bagQuestion4Comment: string;
  bagQuestion5: number;
  bagQuestion5Comment: string;
  bagQuestion6: number;
  bagQuestion6Comment: string;
  bagQuestion7: number;
  bagQuestion7Comment: string;
  bagQuestion8: number;
  bagQuestion8Comment: string;
  bagQuestion9: number;
  bagQuestion9Comment: string;
  bagQuestion10: number;
  bagQuestion10Comment: string;

  // Contact Film
  contactFilmQuestion1: number;
  contactFilmQuestion1Comment: string;
  contactFilmQuestion2: number;
  contactFilmQuestion2Comment: string;
  contactFilmQuestion3: number;
  contactFilmQuestion3Comment: string;
  contactFilmQuestion4: number;
  contactFilmQuestion4Comment: string;
  contactFilmQuestion5: number;
  contactFilmQuestion5Comment: string;
  contactFilmQuestion6: number;
  contactFilmQuestion6Comment: string;
  contactFilmQuestion7: number;
  contactFilmQuestion7Comment: string;
  contactFilmQuestion8: number;
  contactFilmQuestion8Comment: string;
  contactFilmQuestion9: number;
  contactFilmQuestion9Comment: string;
  contactFilmQuestion10: number;
  contactFilmQuestion10Comment: string;

  // Overwrap Film
  overwrapFilmQuestion1: number;
  overwrapFilmQuestion1Comment: string;
  overwrapFilmQuestion2: number;
  overwrapFilmQuestion2Comment: string;
  overwrapFilmQuestion3: number;
  overwrapFilmQuestion3Comment: string;
  overwrapFilmQuestion4: number;
  overwrapFilmQuestion4Comment: string;
  overwrapFilmQuestion5: number;
  overwrapFilmQuestion5Comment: string;
  overwrapFilmQuestion6: number;
  overwrapFilmQuestion6Comment: string;
  overwrapFilmQuestion7: number;
  overwrapFilmQuestion7Comment: string;
  overwrapFilmQuestion8: number;
  overwrapFilmQuestion8Comment: string;
}

File: ClarkSnacks.MTS.Domain/Entities/Inspection.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace ClarkSnacks.MTS.Domain.Entities
{
    public class Inspection
    {
        public Inspection()
        {
            //Lots = new HashSet<Lot>();
        }

        public int Id { get; set; }
        public DateTime DateReceived { get; set; }
        public int VendorId { get; set; }
        public string BOLShipmentNumber { get; set; }
        public bool IsApprovedSupplier { get; set; }
        public int ResultId { get; set; }
        public int DispositionId { get; set; }
        public string DeviationNumber { get; set; }
        public int HoldStatusId { get; set; }
        public string MeasuresToPreventRelease { get; set; }
        public DateTime DateInspected { get; set; }
        public int InspectedById { get; set; }

        // Navigation Properties
        public virtual Operator Operator { get; set; }
        public virtual InspectionQuestion InspectionQuestion { get; set; }
        public virtual Vendor Vendor { get; set; } 
       // public virtual ICollection<Lot> Lots { get; set; }
    }

    public class InspectionQuestion
    {
        public int Id { get; set; }
        public int InspectionId { get; set; }

        public string CartonQuestion1 { get; set; }
        public string CartonQuestion1Comment { get; set; }
        public string CartonQuestion2 { get; set; }
        public string CartonQuestion2Comment { get; set; }
        public string CartonQuestion3 { get; set; }
        public string CartonQuestion3Comment { get; set; }
        public string CartonQuestion4 { get; set; }
        public string CartonQuestion4Comment { get; set; }
        public string CartonQuestion5 { get; set; }
        public string CartonQuestion5Comment { get; set; }
        public string CartonQuestion6 { get; set; }
        public string CartonQuestion6Comment { get; set; }
        public string CartonQuestion7 { get; set; }
        public string CartonQuestion7Comment { get; set; }
        public string CartonQuestion8 { get; set; }
        public string CartonQuestion8Comment { get; set; }

        // Bags
        public string BagQuestion1 { get; set; }
        public string BagQuestion1Comment { get; set; }
        public string BagQuestion2 { get; set; }
        public string BagQuestion2Comment { get; set; }
        public string BagQuestion3 { get; set; }
        public string BagQuestion3Comment { get; set; }
        public string BagQuestion4 { get; set; }
        public string BagQuestion4Comment { get; set; }
        public string BagQuestion5 { get; set; }
        public string BagQuestion5Comment { get; set; }
        public string BagQuestion6 { get; set; }
        public string BagQuestion6Comment { get; set; }
        public string BagQuestion7 { get; set; }
        public string BagQuestion7Comment { get; set; }
        public string BagQuestion8 { get; set; }
        public string BagQuestion8Comment { get; set; }
        public string BagQuestion9 { get; set; }
        public string BagQuestion9Comment { get; set; }
        public string BagQuestion10 { get; set; }
        public string BagQuestion10Comment { get; set; }

        // Contact Film
        public string ContactFilmQuestion1 { get; set; }
        public string ContactFilmQuestion1Comment { get; set; }
        public string ContactFilmQuestion2 { get; set; }
        public string ContactFilmQuestion2Comment { get; set; }
        public string ContactFilmQuestion3 { get; set; }
        public string ContactFilmQuestion3Comment { get; set; }
        public string ContactFilmQuestion4 { get; set; }
        public string ContactFilmQuestion4Comment { get; set; }
        public string ContactFilmQuestion5 { get; set; }
        public string ContactFilmQuestion5Comment { get; set; }
        public string ContactFilmQuestion6 { get; set; }
        public string ContactFilmQuestion6Comment { get; set; }
        public string ContactFilmQuestion7 { get; set; }
        public string ContactFilmQuestion7Comment { get; set; }
        public string ContactFilmQuestion8 { get; set; }
        public string ContactFilmQuestion8Comment { get; set; }
        public string ContactFilmQuestion9 { get; set; }
        public string ContactFilmQuestion9Comment { get; set; }
        public string ContactFilmQuestion10 { get; set; }
        public string ContactFilmQuestion10Comment { get; set; }

        // Overwrap Film
        public string OverwrapFilmQuestion1 { get; set; }
        public string OverwrapFilmQuestion1Comment { get; set; }
        public string OverwrapFilmQuestion2 { get; set; }
        public string OverwrapFilmQuestion2Comment { get; set; }
        public string OverwrapFilmQuestion3 { get; set; }
        public string OverwrapFilmQuestion3Comment { get; set; }
        public string OverwrapFilmQuestion4 { get; set; }
        public string OverwrapFilmQuestion4Comment { get; set; }
        public string OverwrapFilmQuestion5 { get; set; }
        public string OverwrapFilmQuestion5Comment { get; set; }
        public string OverwrapFilmQuestion6 { get; set; }
        public string OverwrapFilmQuestion6Comment { get; set; }
        public string OverwrapFilmQuestion7 { get; set; }
        public string OverwrapFilmQuestion7Comment { get; set; }
        public string OverwrapFilmQuestion8 { get; set; }
        public string OverwrapFilmQuestion8Comment { get; set; }

        public DateTime DateCreated { get; set; }

        // Navigation Properties
        public virtual Inspection Inspection { get; set; }
    }
}
